### PR TITLE
task 6 get /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by HanzoDaBoss**
Complete solution and testing
Complete error handling
Added endpoint to json file

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -109,3 +109,51 @@ describe("/api/articles", () => {
...
+  test("GET 200: responds with an array of comments corresponding to the passed article id", () => {
+    return request(app)
+      .get("/api/articles/9/comments")
+      .expect(200)
+      .then(({body}) => {
+        const {comments} = body;
+        expect(comments.length).toBe(2);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");
+          expect(comment).toHaveProperty("votes");
```
Comment (pippaamy) [inline __tests__/app.test.js:123]: it would be nice to see the value type for each property too! :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -109,3 +109,51 @@ describe("/api/articles", () => {
...
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author");
+          expect(comment).toHaveProperty("body");
+          expect(comment).toHaveProperty("article_id");
+        });
+        expect(comments).toBeSortedBy("created_at", {
```
Comment (pippaamy) [inline __tests__/app.test.js:129]: I would have the sortedBy within its own test!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -109,3 +109,51 @@ describe("/api/articles", () => {
...
+          descending: true,
+          coerce: true,
+        });
+      });
+  });
+  test("GET 404: responds with a status and error message if article id is not found in database", () => {
+    return request(app)
+      .get("/api/articles/100/comments")
+      .expect(404)
+      .then(({body}) => {
```
Comment (pippaamy) [inline __tests__/app.test.js:139]: these error tests are great, fab descriptions too

File: endpoints.json [hunk 4]
```diff
@@ -43,5 +43,21 @@
...
       }
     }
+  },
+  "GET /api/articles/:article_id/comments" : {
+    "description": "served up an array of all comments corresponding to the passed article id",
+    "queries": [],
+    "exampleResponse": {
+      "comments": [
+        {
+          "comment_id": 1,
```
Comment (pippaamy) [inline endpoints.json:53]: well remembered!

File: models/comments-models.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const db = require("../db/connection");
+
+function selectCommentsByArticleId(article_id) {
+  return db
+    .query(
+      `
+    SELECT comment_id, comments.votes, comments.created_at, comments.author, comments.body, articles.article_id FROM comments
+    LEFT JOIN articles ON comments.article_id = articles.article_id
+    WHERE comments.article_id = $1
+    ORDER BY comments.created_at DESC;
```
Comment (pippaamy) [inline models/comments-models.js:10]: this looks great!

File: __tests__/app.test.js [hunk 6]
```diff
@@ -109,3 +109,51 @@ describe("/api/articles", () => {
...
+        expect(body.msg).toBe("Article not found");
+      });
+  });
+  test("GET 404: responds with a status and error message if article id is valid but final pathname is invalid", () => {
+    return request(app)
+      .get("/api/articles/9/invalid-endpoint")
+      .expect(404)
```
Comment (pippaamy) [inline __tests__/app.test.js:146]: this invalid endpoint test may be tested earlier in your topic task?

**Review by pippaamy — COMMENTED** [review]
Great job, I've left some suggestions :)

File: __tests__/app.test.js (hunk 6 above)
Comment (HanzoDaBoss) [reply __tests__/app.test.js:146]: Oh I thought because if article_id is valid, but the path name after is invalid, then it still has to be tested?

**Pull request merged**